augur/filter.py

```diff
@@ -1030,52 +1030,78 @@ def get_groups_for_subsampling(strains, metadata, group_by=None):
             df_dates = pd.DataFrame({col: 'unknown' for col in GROUP_BY_GENERATED_COLUMNS}, index=metadata.index)
             metadata = pd.concat([metadata, df_dates], axis=1)
         else:
-            # Replace date with year/month/day as nullable ints.
-            date_cols = ['year', 'month', 'day']
+            # Create a DataFrame with year/month/day columns as nullable ints.
+            # These columns are prefixed to note temporary usage. They are used
+            # to generate other columns, and will be discarded at the end.
+            temp_prefix = '_augur_filter_'
```

Comment:
It would be safer to add a random string to this prefix. We ask users to rename custom 'year' and 'month' columns in the metadata to use them for grouping. What are the chances they rename it to `_augur_filter_year` 😅

Reply:
Good point, done in [force-push](https://github.com/nextstrain/augur/compare/8375de9f2ef4278c42ed2174019324723060fb43..7eacd26578f8ccab987aabbca112d0f2974c1bf6). Now the chances are near-zero. Maybe if there is a collision we should error and tell them to buy a lottery ticket.